backend/src/packages/users/user-with-password.entity.ts

```diff
@@ -0,0 +1,141 @@
+import { UserEntity } from './user.entity.js';
+
+class UserWithPasswordEntity extends UserEntity {
+  private passwordHash: string;
+
+  private passwordSalt: string;
+
+  private constructor({
+    id,
+    email,
+    fullName,
+    passwordHash,
+    passwordSalt,
+    createdAt,
+    updatedAt,
+    isSurveyCompleted,
+  }: {
+    id: number | null;
+    email: string;
+    fullName: string;
+    passwordHash: string;
+    passwordSalt: string;
+    createdAt: Date | null;
+    updatedAt: Date | null;
+    isSurveyCompleted: boolean;
+  }) {
+    super({
+      id,
+      email,
+      fullName,
+      createdAt,
+      updatedAt,
+      isSurveyCompleted,
+    });
+    this.passwordHash = passwordHash;
+    this.passwordSalt = passwordSalt;
+  }
+
+  public getPasswordInfo(): {
+    passwordHash: string;
+    passwordSalt: string;
+  } {
+    return {
+      passwordHash: this.passwordHash,
+      passwordSalt: this.passwordSalt,
+    };
+  }
+
+  public static initializeWithPassword({
```

Comment:
let's override all methods and name them as in extended model
```suggestion
  public static initialize({
```

Reply:
Done.